lib/iris/_data_manager.py

```diff
@@ -243,18 +247,20 @@ def _realised_dtype_setter(self, realised_dtype):
         if realised_dtype is None:
             self._realised_dtype = None
         else:
-            if not self.has_lazy_data():
-                emsg = 'Cannot set realised dtype, no lazy data is available.'
-                raise ValueError(emsg)
             realised_dtype = np.dtype(realised_dtype)
-            if realised_dtype.kind not in 'biu':
-                emsg = ('Can only cast lazy data to an integer or boolean '
-                        'dtype, got {!r}.')
-                raise ValueError(emsg.format(realised_dtype))
-            self._realised_dtype = realised_dtype
-
-            # Check the manager contract, as the managed dtype has changed.
-            self._assert_axioms()
+            if realised_dtype != self.dtype:
```

Comment:
We only want to set the `_realised_dtype` if it is different from the `self.dtype`. That's pretty important, and I missed porting that into the `DataManager` from the `dask` branch `Cube.dtype` setter ...